dandi/pynwb_utils.py

```diff
@@ -479,14 +480,19 @@ def get_object_id(path: str | Path | Readable) -> Any:
 
 
 def make_nwb_file(
-    filename: StrPath, *args: Any, cache_spec: bool = False, **kwargs: Any
+    filename: StrPath,
+    *args: Any,
+    cache_spec: bool = False,
+    backend: Literal["hdf5", "zarr"] = "hdf5",
```

Comment:
Let's make it more explicit

```suggestion
    backend: Literal["auto", "hdf5", "zarr"] = "auto",
```

with `"auto"` choosing based on extension.